forc-plugins/forc-client/src/util/tx.rs

```diff
@@ -244,6 +249,34 @@ pub(crate) async fn select_secret_key(
     Ok(signing_key)
 }
 
+pub(crate) async fn update_proxy_contract_target(
+    provider: &Provider,
+    secret_key: SecretKey,
+    proxy_contract_id: ContractId,
+    new_target: ContractId,
+) -> Result<()> {
+    abigen!(Contract(
+        name = "ProxyContract",
+        abi = "forc-plugins/forc-client/abi/proxy_contract-abi.json"
+    ));
+
+    let wallet = WalletUnlocked::new_from_private_key(secret_key, Some(provider.clone()));
+
+    let proxy_contract = ProxyContract::new(proxy_contract_id, wallet);
+
+    // TODO: what happens if the call fails? Does 'FuelCallResponse' is returned as Err() in that case?
```

Comment:
We should have a test for this. Is there an "invalid" contract ID we could use to trigger a failure?

Reply:
That is a good point! `async fn test_proxy_contract_re_routes_call()` test in the `tests/deploy.rs` is testing the happy path already. We also have failing test that tests this but specifically tests if the owner is not right. So i think we have both happy and failing tests covered. The problem is non owner test is not calling into this function but does the same thing separetly so we should convert it to call into this function to make sure code coverage itself is also detecting it

Reply:
That's perfect! Done in https://github.com/FuelLabs/sway/pull/6069/commits/17b2bd70e1bd71030545308cffa0d3b15a285cb7